exporter/datadogexporter/internal/logs/translator.go

```diff
@@ -83,6 +88,28 @@ func Transform(lr plog.LogRecord, res pcommon.Resource) datadogV2.HTTPLogItem {
 			l.Message = v.AsString()
 		case "status", "severity", "level", "syslog.severity":
 			status = v.AsString()
+		case "traceid", "contextmap.traceid", "oteltraceid":
+			if traceID, err := decodeTraceID(v.AsString()); err != nil {
+				logger.Error("failed to decode trace id",
+					zap.String("trace_id", v.AsString()),
+					zap.Error(err))
+			} else if l.AdditionalProperties[ddTraceID] == "" {
+				l.AdditionalProperties[ddTraceID] = strconv.FormatUint(traceIDToUint64(traceID), 10)
+				l.AdditionalProperties[otelTraceID] = v.AsString()
+			}
```

Comment:
no need of logging the error. set the properties when err == nil . Avoid if-else as much as possible.